test/fixtures/rules/std_rule_cases/L069.yml

```diff
@@ -0,0 +1,209 @@
+rule: L069
+
+test_pass_cast:
+  pass_str: |
+    select cast(1 as varchar) as bar
+    from foo;
+
+test_pass_casting_operator:
+  pass_str: |
+    select 1::varchar as bar
+    from foo;
+
+test_pass_multi_casting_operator:
+  pass_str: |
+    select 1::int::varchar as bar
+    from foo;
+
+test_pass_convert:
+  pass_str: |
+    select convert(varchar, 1) as bar
+    from foo;
+
+test_pass_3_argument_convert:
+  pass_str: |
+    select convert(varchar, 1, 126) as bar
```

Comment:
Please add another test with the same SQL string but with `preferred_type_casting_style` set to `cast`. In that case, what does the rule  do?
* Generate a lint warning but no autofix
* Pass (no lint warning, no autofix)

Which _should_ it do? I'm thinking the first, but I don't have a strong opinion.

Reply:
It currently does the second but the first one might be better.

Reply:
Implementing the first case is a bit complicated. It is easier when we have just one cast in a statement but in the case of multiple casts, some would be fixable while 3-args convert won't be fixable 
What do you suggest in cases like this:
```
select
        convert(int, 1, 126) as bar,
        100::int::text,
        cast(10 as text) as coo
    from foo;
```

Reply:
I think no fix, just a lint warning. A partial fix would look worse, I think. ❓

Reply:
Hmm, this is harder to really implement as the offending casting style can be anywhere in the statement. We can generate a fix for a previous segment and find an offending segment in the later segments.

Reply:
It's been so long, I can't remember the context on this. What would a partial fix look like in the above case?

Reply:
SQLFluff now allows rules that are called *once* per file, passing it the root parse segment and letting the rule do its own traversal of the file. I don't know if that would be useful in the above case.

Reply:
> SQLFluff now allows rules that are called once per file, passing it the root parse segment and letting the rule do its own traversal of the file. I don't know if that would be useful in the above case.

I am not sure it would help in the case of a statement.

Reply:
Why do the fixed columns appear in a different order? And can you please remind me why they can't all be converted to `cast`? I'm very sorry, but I remember almost no details of this conversation from 45 days ago, even my own original question.

Reply:
For the ordering, there is a different rule doing that, I will have to check for the rule name later.

Yes, I know it's quite long time ago. The original question was about throwing linting error if for instance cast_style is set to cast but since we cannot fix 3 argument convert, we should still throw linting error to the user and the user can decide to fix it manually since we don't auto-fix such. 
Now, I have implemented that but along the way I figured if we have several cast style for example 3 arguments convert and then casting operator while cast-style is set to cast, we fix the casting operator but leave out the convert(that is what I meant by partial fix). Now that I look at it again, partial fix might not have been the right word to use. We should still fix these ones that are fixable and leave anything not fixable out to the user. When we fix the ones matching and a user run lint again, the former convert would/should still raise a lint error.
Let me know my explanation is okay and what you think about my suggestions.
Thanks.

Reply:
Probably L034 is the one changing the column order.

I think I understand now. Each issue found by a lint rule is handled separately. It is reasonable to fix some of these and not others. Often this is done either because there's no reasonable fix, or because autofixing certain issues is more complex and has not been implemented yet. So your description sounds like correct behavior.